pkg/sidecar-controller/snapshot_controller_base.go

```diff
@@ -121,15 +120,12 @@ func NewCSISnapshotSideCarController(
 				// VolumeSnapshotContent.
 				// This will trigger a VolumeSnapshotContent update and it will cause the obj to be re-queued immediately
 				// and CSI CreateSnapshot will be called again without exponential backoff.
+				// Considering the object is modified more than once during the workflow we are not relying on the annoations of oldobj and newobj.
+				// We will just check if newobj status has error and avoid re-queue.
```

Comment:
// Considering the object is modified more than once during the workflow we are not relying on the
// "AnnVolumeSnapshotBeingCreated" annotation. Instead we will just check if newobj status has error
// and avoid the immediate re-queue. This allows the retry to happen with exponential backoff.